lightning/src/chain/mod.rs

```diff
@@ -216,14 +230,10 @@ pub enum ChannelMonitorUpdateStatus {
 /// channel state changes and HTLCs are resolved. See method documentation for specific
 /// requirements.
 ///
-/// Implementations **must** ensure that updates are successfully applied and persisted upon method
-/// completion. If an update will not succeed, then it must immediately shut down.
```

Comment:
`ChannelManager` top level docs need a similar update, they current state that you MUST write each monitor update out to disk before returning from `{watch,update}_channel`.